internal/functions/gerritstatusupdater/gerritstatusupdater.go

```diff
@@ -0,0 +1,386 @@
+		workflowName = *event.Workflow.Name
+
+		// We only care about the start of a workflow or its success
+		switch {
+		case *r.Status == "queued":
+			msg = fmt.Sprintf("Started %s run... see progress at %s", workflowName, *r.HTMLURL)
+		case *r.Status == "completed":
+			// Best-efforts delete build branch regardless of conclusion because
+			// this is the end of the workflow run. If this fails the worst that
+			// will happen is that we build up old build branches. More important
+			// though that we update the CL.
+			if debug {
+				log.Printf("deleting branch %q in repo %q\n", headBranch, repo)
+			}
+			ghclient.Git.DeleteRef(context.Background(), path.Dir(repo), path.Base(repo), "heads/"+headBranch)
+
+			if *r.Conclusion != "success" {
+				return
+			}
+
+			// success
+			msg = fmt.Sprintf("%s run succeeded: %s", workflowName, *r.HTMLURL)
+			result = github.Int(1)
+
+		default:
+			return
+		}
+	case *github.PingEvent:
+		// Handle this to be a good citizen from GitHub's perspective
+		w.Write([]byte("pong"))
+		return
+	default:
+		panic(fmt.Errorf("unhandled event type %T", event))
+	}
+
+	// Build a GerritHub client
+	client, err := fun.buildGerritHubClient(repo, workflowPath)
+	if err != nil {
+		panic(err)
+	}
+	if client == nil {
+		if debug {
+			log.Printf("no configuration specified for repo-workflowPath %q\n", buildRepoWorkflowPathKey(repo, workflowPath, ""))
+		}
+		return
+	}
+
+	// Establish changeID and revisionID from head branch. Note
+	// the first part of the build branch is not significant
+	// as far as gerritstatusupdater is concerned: it's simply
+	// used to namespace build branches, but also for workflows
+	// to conditionally run based on the first part.
+	headBranchParts := strings.Split(headBranch, "/")
+	if len(headBranchParts) != 3 {
+		panic(fmt.Errorf("head branch %q not in expected format", headBranch))
+	}
+	changeID, revisionID := headBranchParts[1], headBranchParts[2]
+
+	ri := &gerrit.ReviewInput{
+		Tag:     strings.ToLower(workflowName),
+		Message: msg,
+	}
+	if result != nil {
+		ri.Labels = map[string]string{
+			workflowName + "-Result": strconv.Itoa(*result),
+		}
+	}
+
+	if !debug {
+		if _, _, err := client.Changes.SetReview(changeID, revisionID, ri); err != nil {
+			panic(fmt.Errorf("failed to update gerrit: %v", err))
+		}
+	} else {
+		b, _ := json.MarshalIndent(ri, "", "  ")
+		log.Printf("gerrit.SetReview %s/%s with\n%s\n", changeID, revisionID, b)
+	}
+}
+
+// buildGitHubClient returns a GitHub client, deriving authentication for the
+// given repo from the environment. If environment configuration cannot be
+// found for repo, an error is returned (because in this case
+// gerritstatusupdater should not be configured to listen to that repo).
+func (fun Function) buildGitHubClient(repo string) (*github.Client, error) {
+	// Lookup GitHub access token from the env of this repo
+	githubPATKey := buildRepoKey(repo, EnvGitHubPATSuffix)
+	githubPAT, ok := os.LookupEnv(githubPATKey)
```

Comment:
nit: use Getenv and check whether the value is empty. right now, if it is set but empty, you will error in a different way, presumably for no reason.

Reply:
Excellent catch, thanks. Done